packages/manager/apps/pci-savings-plan/src/hooks/useSavingsPlan.tsx

```diff
@@ -198,3 +200,13 @@ export const useSavingsPlanContract = () => {
     queryFn: () => getSavingsPlanContracts(serviceId),
   });
 };
+
+export const useSavingsPlanId = (): string => {
```

Comment:
As you are using the same hook for another param, I'll suggest to add some generic custom hook, something like that:

```ts
const useParam = (param: string): string => {
  const { [param]: paramValue } = useParams();
  if (!paramValue) {
    throw new Error(`Missing ${param} in URL.`);
  }
  return paramValue;
}
```

Reply:
@lionel95200x For sure 👍🏻